tests/unit/test_charm.py

```diff
@@ -3395,10 +3417,180 @@ def test_given_certificate_rejected_in_notary_when_configure_then_certificate_re
                     "is_initialized.return_value": True,
                     "token_is_valid.return_value": True,
                     "get_certificate_requests_table.return_value": CertificateRequests(
-                        rows=[CertificateRequest(id=1, csr=str(csr), certificate_chain="rejected")]
+                        rows=[
+                            CertificateRequestRow(id=1, csr=str(csr), certificate_chain="rejected")
+                        ]
                     ),
                 },
             ),
         ):
             context.run(context.on.update_status(), state)
         mock_set_relation_certificate.assert_called_once()
+
+    @patch(f"{TLS_LIB_PATH}.TLSCertificatesRequiresV4.get_assigned_certificate")
+    def test_given_access_relation_created_when_configure_then_certificate_not_replaced(
+        self, mock_assigned_certificates, context
+    ):
+        with tempfile.TemporaryDirectory() as tempdir:
+            config_mount = Mount(location="/etc/notary/config", source=tempdir)
+            state = State(
+                storages={Storage(name="config"), Storage(name="database")},
+                containers=[
+                    Container(
+                        name="notary",
+                        can_connect=True,
+                        mounts={"config": config_mount},
+                        layers={
+                            "notary": Layer(
+                                {
+                                    "summary": "notary layer",
+                                    "description": "pebble config layer for notary",
+                                    "services": {
+                                        "notary": {
+                                            "override": "replace",
+                                            "summary": "notary",
+                                            "command": "notary -config /etc/notary/config/config.yaml",
+                                            "startup": "enabled",
+                                        }
+                                    },
+                                }
+                            )
+                        },
+                    )
+                ],
+                relations=[Relation(id=1, endpoint=TLS_ACCESS_RELATION_NAME)],
+                leader=True,
+            )
+            certificate, _ = self.example_cert_and_key()
+            with open(tempdir + "/certificate.pem", "w") as f:
+                f.write(str(certificate))
+            mock_assigned_certificates.return_value = (None, None)
+            with patch(
+                "notary.Notary.__new__",
```

Comment:
I think I remember reading somewhere that patching `__new__` is dangerous and can lead to some unexpected / undefined behaviour.

I was working around this by patching the class (`notary.Notary`) with a mock that returned another mock (i.e. it had a `return_value` that was a mock itself). I admit it isn't pretty, but that is what the pytest authors seemed to encourage. I don't know where I read that though, so do with that information what you'd like!

Reply:
I will choose to ignore it, because it took me a ton of time to get this to work :D